connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/util/IntegrationTestUtils.scala

```diff
@@ -57,6 +57,15 @@ object IntegrationTestUtils {
     Files.exists(Paths.get(filePath))
   }
 
+  private[sql] lazy val isCatalystTestJarAvailable: Boolean = {
```

Comment:
I want to know which scenarios `isCatalystTestJarAvailable` may be false now

Reply:
Build without test jar?

Reply:
I test the following scenarios:

1. build/sbt clean "connect-client-jvm/test"         
2. build/sbt clean "connect-client-jvm/testOnly *ClientE2ETestSuite"

3. 
```
build/mvn clean
build/mvn clean package test -Dtest=none -DwildcardSuites=org.apache.spark.sql.ClientE2ETestSuite
```
4.
```
build/mvn clean
build/mvn clean install -pl connector/connect/server -am -DskipTests   
build/mvn clean install -pl assembly -DskipTests
build/mvn clean test -pl connector/connect/client/jvm
``` 

New tests can be triggered and tested successfully. 

@Hisoka-X So haver you thought of any other way to successfully test before this PR, but without `isCatalystTestJarAvailable ` checking, the test will fail?

Reply:
Before add `isCatalystTestJarAvailable` check, I didn't modify sbt build script. So it didn't work on sbt . Check https://github.com/apache/spark/pull/40564#discussion_r1152081937 . But now I think I can remove this check.